// fashion/test/helpers.js
var Fashion = require('../index.js');
var assert = require('assert');

var testBody = function(scss, css, doDebug, before, after, done, file){
        var scssCode = scss.join('\n'),
            builder = new Fashion.Builder();

        before && before(builder);
        var cssOut = builder.compile(scssCode, file),
            expected = css.join('\n');

        cssOut.getText(function(generated){
            if(doDebug) {
                debugger;
            }
            assert.equal(generated.join(''), expected);
            after && after(builder);
            done && done();
        });

    },
    test = function (desc, scss, css, doDebug, before, after, file) {
        it(desc, function (done) {
            testBody(scss, css, doDebug, before, after, done, file);
        });
    },
    testOnly = function (desc, scss, css, doDebug, before, after, file) {
        it.only(desc, function (done) {
            testBody(scss, css, doDebug, before, after, done, file);
        });
    },
    testFailure = function (desc, scss, exception, doDebug, before, after) {
        it(desc, function () {
            try {
                testBody(scss, [], doDebug, before, after);
                fail("Expected exception with message : " + exception);
            } catch (ex) {
                assert(ex.message.indexOf(exception) > -1);
            }
        });
    },
    xtest = function (desc, scss, css) {
        xit(desc, function () {
        });
    };

module.exports = {
    testBody: testBody,
    test: test,
    testOnly: testOnly,
    testFailure: testFailure,
    xtest: xtest
}
